### PR TITLE
[RELAY] Filter PlaceholderOp from schedule.

### PR DESCRIPTION
**Pull request by srkreddy1238**
Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/dmlc/tvm/blob/master/CONTRIBUTORS.md#reviewers).

### CODE REVIEW

**Comment by tqchen** [comment]
This does not look right, what is the motivation on this? The compile engine should have included all the inputs including placeholder

**Comment by srkreddy1238** [comment]
this is to handle a situation like below

```
--------------------------
fn (%p0: Tensor[(1, 10000), float32],
    %p1: Tensor[(1, 2, 1, 200), float32],
    %p2: Tensor[(1, 2, 1, 200), float32])
    -> Tuple[Tensor[(1, 10000), float32], Tensor[(2, 2, 1, 200), float32]] {
  %0 = (%p1, %p2)
  %1 = concatenate(%0) # ty=Tensor[(2, 2, 1, 200), float32]
  %2 = (%p0, %1)
  %2
}
--------------------------
```

where p0 is a bypass. Ref. https://github.com/dmlc/tvm/pull/2216#issuecomment-452786342

**Comment by tqchen** [comment]
OK, this seems to be fine, can you please document the intent of the code as a  comment block?

**Comment by srkreddy1238** [comment]
Test case added, but enabling fuse pass has another issue as given below.

```
Traceback (most recent call last):
  File "./tests/python/relay/test_backend_compile_engine.py", line 48, in <module>
    test_compile_placeholder_bypass()
  File "./tests/python/relay/test_backend_compile_engine.py", line 44, in test_compile_placeholder_bypass
    graph, lib, params = relay.build(func, 'llvm')
  File "/home/srk/.local/lib/python3.6/site-packages/tvm-0.5.dev0-py3.6-linux-x86_64.egg/tvm/relay/build_module.py", line 276, in build
    lowered_funcs, target=target, target_host=target_host)
  File "/home/srk/.local/lib/python3.6/site-packages/tvm-0.5.dev0-py3.6-linux-x86_64.egg/tvm/build_module.py", line 592, in build
    mhost = codegen.build_module(fhost_all, str(target_host))
  File "/home/srk/.local/lib/python3.6/site-packages/tvm-0.5.dev0-py3.6-linux-x86_64.egg/tvm/codegen.py", line 20, in build_module
    return _Build(lowered_func, target)
  File "/home/srk/.local/lib/python3.6/site-packages/tvm-0.5.dev0-py3.6-linux-x86_64.egg/tvm/_ffi/_ctypes/function.py", line 185, in __call__
    ctypes.byref(ret_val), ctypes.byref(ret_tcode)))
  File "/home/srk/.local/lib/python3.6/site-packages/tvm-0.5.dev0-py3.6-linux-x86_64.egg/tvm/_ffi/base.py", line 72, in check_call
    raise TVMError(py_str(_LIB.TVMGetLastError()))
tvm._ffi.base.TVMError: [11:53:12] /home/srk/work/DMLC/tvm/src/codegen/llvm/llvm_module.cc:173: LLVM module verification failed with the following errors: 
Invalid operand types for FCmp instruction
  %243 = fcmp oeq i8* %34, %242
```

I may need some help here.

**Comment by jroesch** [comment]
This makes sense to me, we need someone with more TVM experience to explain the LLVM failure. It looks like the compiler is producing invalid IR.

File: src/relay/backend/compile_engine.cc [hunk 1]
```diff
@@ -83,11 +83,18 @@ class ScheduleGetter :
     cache_node->func_name = readable_name_stream_.str();
     CachedFunc cfunc(cache_node);
     CHECK(master_op_.defined());
+    // Filter PlaceholderOp from the outputs for schedule.
```
Comment (jroesch) [inline src/relay/backend/compile_engine.cc]: Can we add a more descriptive comment here? for example, explaining that fusion over tupled results may leave identity relationships between inputs and outputs, and those should not be scheduled.

**Comment by srkreddy1238** [comment]
cc @tqchen

**Comment by tqchen** [comment]
Thanks to @srkreddy1238 @jroesch , this is now merged

**Pull request merged**